gqlalchemy/memgraph.py

```diff
@@ -585,3 +585,50 @@ def get_lambda(expression: str) -> str:
             return ""
         else:
             return f"(e, v | {expression})"
+
+
+class BreadthFirstSearch(IntegratedAlgorithm):
+    """Build a BFS call for a Cypher query.
+
+    The Breadth-First Search can be called in Memgraph with Cypher queries
+    such as:
+    MATCH (a {id: 723})-[*BFS ..10 (e, v | e.x > 12 AND v.y < 3)]-() RETURN *;
+    It is called inside the relationship clause, "* BFS" naming the algorithm,
+    "..10" specifying depth bounds, and "(e, v | <expression>)" is a filter
+    lambda.
+    """
+
+    def __init__(
+        self,
+        lower_bound: int = None,
+        upper_bound: int = None,
+        condition: str = None,
```

Comment:
Have you thought about creating a `Condition` class for this? So that we don't have `str`, but `Condition` type which has left and right expression with the logical operator between them. This is just a suggestion, maybe for the next releases.

Reply:
This is a good suggestion, I will write it down and try to implement it after we merge this, wShortest and DFS PR's into the same branch. Is that OK?